Issue125 compiler specific flags

mikebentley15 opened the pull request:
Fixes #125 

**Description:**
We wanted a way to specify compiler flags specific to only one of the compilers and not all of them.  In this pull request, I added two settings for each compiler in the `flit-config.toml` file.

- `fixed_compile_flags`: compile flags for only this compiler when compiling object files and will also be used during the link stage.
- `fixed_link_flags`: link-time flags for only this compiler when linking

Currently, these are tied to the compiler _type_ instead of the compiler _name_, and therefore we still only support one of each supported compiler types.  We hope to change that with issue #40.

The flags placed into these two settings get put into the generated `Makefile`.  For a `gcc` type compiler, these variables are `GCC_CXXFLAGS` and `GCC_LDFLAGS` respectively.

**Question**
Do we want to remove the automatic insertion of the `--gcc-toolchain` flag for Clang compilers?  Now that it is documented, we could expect users to specify that flag.  Currently, if the user wants to specify a different `libstdc++` version to use than the GCC compiler specified in `flit-config.toml`, then there is no clear way to do that.

However, if we were to do such a change, we could potentially break backward functionality compatibility and cause successful uses of flit to no longer work without fixing `flit-config.toml`.  Also, it may make FLiT harder to use since you now have to worry about which standard library your Clang and Intel compilers are using.  Granted, I think it's helpful to understand what's going on, I also understand many people don't have the time for that.

@IanBriggs what do you think?  Should we remove it and expect the user to be more explicit and hope s/he reads the manual?  Or should we keep it for ease of use and for backward compatibility?  I really don't like that there is no way to change the automatically inserted `--gcc-toolchain` flag.

**Documentation:**

- I added to the page documenting the flit configuration file.
- I created a new page dedicated to C++11 standard library issues and how to use this functionality to get around it, as well as examples of flags to specify a specific C++11 standard library (non-exhaustive)

**Tests:**
I added two test cases

- `flit_cli/flit_update/tst_compilerspecificflags.py`:  tests that the settings in `flit-config.toml` make it into the appropriate variables within the generated `Makefile`
- `flit_makefile/tst_build_with_compiler_specific_flags.py`: tests that the generated Makefile actually uses the flags during commpilations of source files

IanBriggs commented:
I say we should break backwards compatibility in this case.
It should be safe to assume that users wanting to mix compilers should know how, and if they don't we have documentation.

mikebentley15 commented:
Thanks for the input.  I agree.  Besides, if we have any users besides us, I don't know about them...

mikebentley15 commented:
I've removed the insertion of the `--gcc-toolchain` flag from the generated `Makefile` and updated documentation accordingly.  I also updated tests for this change.
